analyzer/linux/lib/core/packages.py

```diff
@@ -157,38 +158,55 @@ def finish(self):
     def get_pids(self):
         return []
 
-    def apicalls_analysis(self):
+    def strace_analysis(self):
         kwargs = {"args": self.args, "timeout": self.timeout, "run_as_root": self.run_as_root}
         log.info(self.target)
-        cmd = apicalls(self.target, **kwargs)
-        stap_start = timeit.default_timer()
+        
+        target_cmd = f'{self.target}'
+        if "args" in kwargs:
+            target_cmd += f' {" ".join(kwargs["args"])}'
+
+        cmd = f"sudo strace -ttffn -o {self.strace_output}/strace.log {target_cmd}"
         log.info(cmd)
         self.proc = subprocess.Popen(
             cmd, env={"XAUTHORITY": "/root/.Xauthority", "DISPLAY": ":0"}, stderr=subprocess.PIPE, shell=True
         )
-
-        while b"systemtap_module_init() returned 0" not in self.proc.stderr.readline():
-            # log.debug(self.proc.stderr.readline())
-            pass
-
-        stap_stop = timeit.default_timer()
-        log.info("Process startup took %.2f seconds", stap_stop - stap_start)
+        log.info("Process started with strace")
         return True
 
+    ## This feature has already been migrated to Auxiliary:stap.py
+    # def apicalls_analysis(self):
+    #     kwargs = {"args": self.args, "timeout": self.timeout, "run_as_root": self.run_as_root}
+    #     log.info(self.target)
+    #     cmd = apicalls(self.target, **kwargs)
+    #     stap_start = timeit.default_timer()
+    #     log.info(cmd)
+    #     self.proc = subprocess.Popen(
+    #         cmd, env={"XAUTHORITY": "/root/.Xauthority", "DISPLAY": ":0"}, stderr=subprocess.PIPE, shell=True
+    #     )
+
+    #     while b"systemtap_module_init() returned 0" not in self.proc.stderr.readline():
+    #         # log.debug(self.proc.stderr.readline())
+    #         pass
+
+    #     stap_stop = timeit.default_timer()
+    #     log.info("Process startup took %.2f seconds", stap_stop - stap_start)
+    #     return True
+
```

Comment:
probably worth gutting any systemtap-based support in the analyzer at this point

Reply:
Even the auxiliary module [stap.py](https://github.com/kevoreilly/CAPEv2/blob/master/analyzer/linux/modules/auxiliary/stap.py)?

Reply:
That looks great!

Years ago SystemTap would have been a go-to tool for a lot of use cases, though it had its limitations. In the meantime eBPF has taken over, replacing nearly every use case I can think of for SystemTap, and doing it better.